src/keypay-dotnet/Auth/OAuth2AuthorizationRequestHeaderAuthenticator.cs

```diff
@@ -0,0 +1,48 @@
+using System;
+using System.Linq;
+using RestSharp;
+using RestSharp.Authenticators;
+
+namespace KeyPay.Auth
+{
+    /// <summary>
+    /// The OAuth 2 authenticator using the authorization request header field.
+    /// </summary>
+    /// <remarks>
+    /// Based on http://tools.ietf.org/html/draft-ietf-oauth-v2-10#section-5.1.1
+    /// </remarks>
+    public class OAuth2AuthorizationRequestHeaderAuthenticator : RestSharp.Authenticators.OAuth2.OAuth2AuthorizationRequestHeaderAuthenticator
+    {
+        /// <summary>
+        /// Stores the Authorization header value as "[tokenType] accessToken". used for performance.
+        /// </summary>
+        private readonly string authorizationValue;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="OAuth2AuthorizationRequestHeaderAuthenticator"/> class.
+        /// </summary>
+        /// <param name="accessToken">
+        /// The access token.
+        /// </param>
+        public OAuth2AuthorizationRequestHeaderAuthenticator(string accessToken)
```

Comment:
I'm not sure about this one. Should the token type be `"Bearer"` rather than `"OAuth"`? Do we even need this class or can we just initialise a `OAuth2AuthorizationRequestHeaderAuthenticator` with the access token and `"Bearer`" token type?

Reply:
I actually copied that from KP codebase but now looking at the same class from RestSharp, we don't need it anymore. So I have changed to use the one from RestSharp instead (reference code [here](https://github.com/restsharp/RestSharp/blob/dev/src/RestSharp/Authenticators/OAuth2/OAuth2AuthorizationRequestHeaderAuthenticator.cs))